### PR TITLE
Generate API docs

### PR DESCRIPTION
**Pull request by JoshVanL**
**What this PR does / why we need it**:
Adds Scripts to generate API documentation for the site and makes sure it's up to date in CI.

**Special notes for your reviewer**:
Many API type definitions violated the generators validations so have been changed to follow convention. 

```release-note
Generate API documentation in site
```
/hold
/assign

### CODE REVIEW

File: docs/generated/reference/manifest.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (munnerz) [inline docs/generated/reference/manifest.json]: This manifest.json file shouldn't be committed (at least, we don't commit it in either the docker based or bazel based docgen pipeline)

File: docs/generated/reference/includes/_definitions.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (munnerz) [inline docs/generated/reference/includes/_definitions.md]: We also don't commit the `includes/` dir (we only commit static_includes)

**Comment by JoshVanL** [comment]
/unassign
/assign @simonswine 
Waiting on https://github.com/jetstack/testing/pull/87

**Comment by munnerz** [comment]
Added a comment to the testing PR - didn't notice that yesterday sorry!

**Comment by simonswine** [comment]
/assign @JoshVanL 
/unassign

needs rebase

File: Makefile [hunk 3]
```diff
@@ -161,7 +161,7 @@ verify_codegen:
 	$(HACK_DIR)/verify-codegen.sh
 
 verify_vendor: $(BINDIR)/dep
-	dep ensure -no-vendor -dry-run -v
+	$(BINDIR)/dep ensure -no-vendor -dry-run -v
```
Comment (simonswine) [inline Makefile]: Why are we doing that? `PATH` contain `$(BINDIR)`

**Comment by JoshVanL** [comment]
/unassign
/assign @simonswine 
@simonswine Not using the new image? nodejs should install npm

**Pull request closed without merging**